sqlmesh/core/context.py

```diff
@@ -1372,42 +1372,46 @@ def test(
         else:
             verbosity = 1
 
-        try:
-            if tests:
-                result = run_model_tests(
-                    tests=tests,
-                    models=self._models,
-                    engine_adapter=self._test_engine_adapter,
-                    dialect=self.default_dialect,
-                    verbosity=verbosity,
-                    patterns=match_patterns,
-                    preserve_fixtures=preserve_fixtures,
-                    default_catalog=self.default_catalog,
-                )
-            else:
-                test_meta = []
-
-                for path, config in self.configs.items():
-                    test_meta.extend(
-                        get_all_model_tests(
-                            path / c.TESTS,
-                            patterns=match_patterns,
-                            ignore_patterns=config.ignore_patterns,
-                        )
+        if tests:
+            result = run_model_tests(
+                tests=tests,
+                models=self._models,
+                engine_adapter=self._test_engine_adapter,
+                config=self.config,
+                gateway=self.gateway,
+                dialect=self.default_dialect,
+                verbosity=verbosity,
+                patterns=match_patterns,
+                preserve_fixtures=preserve_fixtures,
+                stream=stream,
+                default_catalog=self.default_catalog,
+                default_catalog_dialect=self.engine_adapter.DIALECT,
+            )
+        else:
+            test_meta = []
+
+            for path, config in self.configs.items():
+                test_meta.extend(
+                    get_all_model_tests(
+                        path / c.TESTS,
+                        patterns=match_patterns,
+                        ignore_patterns=config.ignore_patterns,
                     )
-
-                result = run_tests(
-                    test_meta,
-                    models=self._models,
-                    engine_adapter=self._test_engine_adapter,
-                    dialect=self.default_dialect,
-                    verbosity=verbosity,
-                    preserve_fixtures=preserve_fixtures,
-                    stream=stream,
-                    default_catalog=self.default_catalog,
                 )
-        finally:
-            self._test_engine_adapter.close()
+
+            result = run_tests(
+                model_test_metadata=test_meta,
+                models=self._models,
+                engine_adapter=self._test_engine_adapter,
```

Comment:
Why even pass the engine_adapter here? Shouldn't we figure the default one out downstream?

Reply:
Yeah I was thinking about this as well. The reason I did this was to avoid re-instantiating the test engine adapter. We instantiate it once in the `Context` constructor and use it in a few other places besides `Context.test`, so I couldn't just move it in `run_tests` too.

If that's ok I'd be happy to move it here.

Reply:
To be fair the only other place it's used is in the `info` command. Provided that tests may refer to multiple adapters, the usefulness of testing the test connection as part of the `info` command becomes somewhat less clear. Also users can just run tests to test the connection.

Should we move the test connection management out of the context altogether?

Reply:
Yep, this seems reasonable to me.

One gotcha is that `create_test` also [uses the testing adapter](https://github.com/TobikoData/sqlmesh/blob/f0ae0a189d63a91f179cc30e9d9bf9211a49e3b2/sqlmesh/core/context.py#L1351), but that shouldn't be a problem I think. We could move the test connection management in `run_tests` and then separately create on-demand the testing adapter for `create_test` in its body, making sure to close the connection after it's done. Wdyt?

Reply:
Yes, I like this

Reply:
Gotcha thanks, will refactor

Reply:
Updated, added docs as well.